struct.c

```diff
@@ -297,6 +300,30 @@ rb_struct_s_inspect(VALUE klass)
     return inspect;
 }
 
+static VALUE
+rb_data_s_new(int argc, const VALUE *argv, VALUE klass)
+{
+    if(rb_keyword_given_p()) {
+        if (argc > 1 || !RB_TYPE_P(argv[0], T_HASH)) {
+            rb_raise(rb_eArgError, "wrong number of arguments (given %d, expected 0)", argc);
+        }
+        return rb_class_new_instance_pass_kw(argc, argv, klass);
+    } else {
+        VALUE members = struct_ivar_get(klass, id_members);
+        int num_members = RARRAY_LEN(members);
+
+        if (num_members < argc) {
+            rb_raise(rb_eArgError, "wrong number of arguments (given %d, expected %d)", argc, num_members);
+        }
```

Comment:
Since `Data` instances don't have writer methods, insufficient arguments shouldn't be an error too?

```suggestion
        rb_check_arity(argc, num_members, num_members);
```

Reply:
@nobu It is handled in `initialize`, so the inherited class can provide the defaults. From examples: 
```ruby
Measure = Data.define(:amount, :unit) do
 NONE = Data.define

 def initialize(amount:, unit: NONE.new)
   super(amount: Float(amount), unit:)
 end
end

Measure.new('10', 'km') # => #<data Measure amount=10.0, unit="km">
Measure.new(10_000)     # => #<data Measure amount=10000.0, unit=#<data NONE>>
```

Reply:
Makes sense, then?

```suggestion
         rb_check_arity(argc, 0, num_members);
```

This is tolerant for the message change (if it were happen).